data/pins.yml

```diff
@@ -0,0 +1,437 @@
+# UARTs
+ser0_tx:
+  block_ios:
+    - block:    uart
+      instance: 0
+      io:       tx
+ser0_rx:
+  block_ios:
+    - block:    uart
+      instance: 0
+      io:       rx
+ser1_tx:
+  block_ios:
+    - block:    uart
+      instance: 1
+      io:       tx
+ser1_rx:
+  block_ios:
+    - block:    uart
+      instance: 1
+      io:       rx
+rs232_tx:
+  block_ios:
+    - block:    uart
+      instance: 4
+      io:       tx
+rs232_rx:
+  block_ios:
+    - block:    uart
+      instance: 4
+      io:       rx
+
+# QWIIC (Sparkfun) buses
+scl0:
+  block_ios:
+    - block:    i2c
+      instance: 0
+      io:       scl
+sda0:
+  block_ios:
+    - block:    i2c
+      instance: 0
+      io:       sda
+scl1:
+  block_ios:
+    - block:    i2c
+      instance: 1
+      io:       scl
+sda1:
+  block_ios:
+    - block:    i2c
+      instance: 1
+      io:       sda
+
+# SPI LCD
+lcd_copi:
+  block_ios:
+    - block:    spi
+      instance: 0
+      io:       tx
+# RX or CIPO is not connected
+lcd_clk:
+  block_ios:
+    - block:    spi
+      instance: 0
+      io:       sck
+
+# SPI Flash interface
+appspi_d0:
+  block_ios:
+    - block:    spi
+      instance: 1
+      io:       tx
+appspi_d1:
+  block_ios:
+    - block:    spi
+      instance: 1
+      io:       rx
+appspi_clk:
+  block_ios:
+    - block:    spi
+      instance: 1
+      io:       sck
+
+# SPI Ethernet
+ethmac_copi:
+  block_ios:
+    - block:    spi
+      instance: 2
+      io:       tx
+ethmac_cipo:
+  block_ios:
+    - block:    spi
+      instance: 2
+      io:       rx
+ethmac_sclk:
+  block_ios:
+    - block:    spi
+      instance: 2
+      io:       sck
+
+# Raspberry Pi HAT
+rph_g0:
```

Comment:
This setup is fine for getting things going but for the v1.0 release I do want to enable more flexibility, e.g. so a given set of rpi HAT pins could be GPIO, could be I2C, could be UART or could be SPI. If you restrict yourself to what the rpi can actually do (uncertain how flexible it is in connecting its own SPI etc blocks to the GPIO pins) you may have a more restricted set but it is also a general purpose header people can connect non RPI HAT stuff to so flexibility is useful.

Reply:
I've captured this here: https://github.com/lowRISC/sonata-system/issues/169